### DISCUSSION TITLE
I have a great idea.

### DISCUSSION THREAD

**Discussion by Mr-Programmer-777** [ScratchAddons/ScratchAddons#1798] [Ideas]
DRUMROLL!!!

An addon creator!

How it would work is by creating a place where you can get unofficial addons.. Not just the ones added by @TheColaber or others, but a marketplace! Additionally, we could branch the scratch editor and re-model it into a JS editor or the like. e.g; you can drag 'n drop a button into a replica of the scratch website, set a place for it, edit it, add some text, code and graphics, and... POOF! You've got your own addon! This would be a flippin' TON of work (i know..) but...
IT.
WOULD.
BE. 
AMAZING!!!
Think about it!

I would be free to help with graphics as much as possible.
Brainstorm your ideas here in the comments while I blueprint up a storm!

Yours excitedly,
MrProgrammer777

Edit: I'm going to start learning JS! Im using w3schools.com, but feel free to recommend any other site!

**Comment by cobaltt7** [comment]
Rejected, chrome webstore reviews won’t like it

**Reply by Mr-Programmer-777** [reply]
Explain..

**Reply by cobaltt7** [reply]
plus it's not really possible, see #287

**Reply by Mr-Programmer-777** [reply]
> Rejected, chrome webstore reviews won’t like it

Why?

**Reply by cobaltt7** [reply]
See #287

**Reply by Mr-Programmer-777** [reply]
I did, but it doesn't explain WHY people don't like it..

**Reply by TheColaber** [reply]
It is not about why people don't like it, It is rejected by the chrome reviewers. As well as the point W_L made:
> Scratch Addons isn't an addon manager. Those are Tampermonkey and Stylus.

**Reply by Mr-Programmer-777** [reply]
Aww. So do the chrome reviewers have to give the thumbs up first? Additionally, do they not like marketplaces or something?

**Reply by mxmou** [reply]
> Aww. So do the chrome reviewers have to give the thumbs up first? Additionally, do they not like marketplaces or something?

Yes, on Chrome all updates have to get reviewed. Marketplaces in general aren't a problem, the problem is running code that isn't part of the extension (because it can't be guaranteed that it's safe).

**Comment by Mr-Programmer-777** [comment]
@mxmou 
> > Aww. So do the chrome reviewers have to give the thumbs up first? Additionally, do they not like marketplaces or something?
> 
> Yes, on Chrome all updates have to get reviewed. Marketplaces in general aren't a problem, the problem is running code that isn't part of the extension (because it can't be guaranteed that it's safe).

Right. Can't we make it like scratch? Like an actual website, as I said, but then someone like @TheColaber or @WorldLanguages can review created addons by the community, and decide which ones to implement next?
I see that as a good idea.

**Reply by Mr-Programmer-777** [reply]
Then, it can be guaranteed safe, and be tagged Community in the Settings of SA.

**Reply by mxmou** [reply]
This would be possible but the community can already contribute addons so I'm not sure why we would want to make this.

**Reply by TheColaber** [reply]
Plus, how much of the community *can* make addons? Not even half the people that work on this project can...

**Reply by Mr-Programmer-777** [reply]
It would be a really cool, easy and fun way to make and share addons. Plus, it gets the whole community chiming in; ~10.5k people making addons!

**Reply by TheColaber** [reply]
If you learn to make addons, we can consider this. All jokes aside, this just can't happen properly. Maybe only 20 of those people will actually know how scratch works, and how to program in JS,
